flopy/mf6/data/mfdataplist.py

```diff
@@ -0,0 +1,2527 @@
+                            "Storing {} to external file {}.."
+                            ".".format(self.structure.name, external_file_path)
+                        )
+                    external_data = {
+                        "filename": external_file_path,
+                        "data": data,
+                        "binary": binary,
+                    }
+                    self._set_record(external_data, check_data=check_data)
+
+    def external_file_name(self):
+        """Returns external file name, or None if this is not external data."""
+        storage = self._get_storage()
+        if storage is None:
+            return None
+        if (
+            storage.data_storage_type == DataStorageType.external_file
+            and storage.fname is not None
+            and storage.fname != ""
+        ):
+            return storage.fname
+        return None
+
+    @staticmethod
+    def _file_data_to_memory(fd_data_file, first_line):
+        """
+        scan data file from starting point to find the extent of the data
+
+        Parameters
+        ----------
+        fd_data_file : file descriptor
+            File with data to scan.  File location should be at the beginning
+            of the data.
+
+        Returns
+        -------
+        list, str : data from file, next line in file after data
+        """
+        data_lines = []
+        clean_first_line = first_line.strip().lower()
+        if clean_first_line.startswith("end"):
+            return data_lines, fd_data_file.readline()
+        if len(clean_first_line) > 0 and clean_first_line[0] != "#":
+            data_lines.append(clean_first_line)
+        line = fd_data_file.readline()
+        while line:
+            line_mod = line.strip().lower()
+            if line_mod.startswith("end"):
+                return data_lines, line
+            if len(line_mod) > 0 and line_mod[0] != "#":
+                data_lines.append(line_mod)
+            line = fd_data_file.readline()
+        return data_lines, ""
+
+    def _dataframe_check(self, data_frame):
+        valid = data_frame.shape[0] > 0
+        if valid:
+            for name in self._header_names:
+                if (
+                    name != "boundname"
+                    and data_frame[name].isnull().values.any()
+                ):
+                    valid = False
+                    break
+        return valid
+
+    def _try_pandas_read(self, fd_data_file):
+        delimiter_list = [" ", "\t", ","]
+        for delimiter in delimiter_list:
+            try:
+                # read flopy formatted data, entire file
+                data_frame = pandas.read_csv(
+                    fd_data_file,
+                    sep=delimiter,
+                    names=self._header_names,
+                    dtype=self._data_header,
+                    comment="#",
+                    index_col=False,
+                    skipinitialspace=True,
+                )
```

Comment:
It might for some cases, but for other cases it could be a problem.  I decided to keep the delimiter list, but added "\\s+" to the list which according to the documentation is functionally equivalent to delim_whitespace=True.